stdlib/public/core/UnicodeScalar.swift

```diff
@@ -64,18 +64,29 @@ public struct UnicodeScalar :
 
   /// Creates a Unicode scalar with the specified numeric value.
   ///
+  /// - Parameter v: The Unicode code point to use for the scalar. `v` must be
+  ///   a valid Unicode scalar value, in the range `0...0xD7FF` or
+  ///   `0xE000...0x10FFFF`. In case of an invalid unicode scalar value, nil is
+  ///   returned.
+  ///
   /// For example, the following code sample creates a `UnicodeScalar` instance
   /// with a value of an emoji character:
   ///
   ///     let codepoint: UInt32 = 127881
   ///     let emoji = UnicodeScalar(codepoint)
-  ///     print(emoji)
+  ///     print(emoji!)
   ///     // Prints "🎉"
   ///
-  /// - Parameter v: The Unicode code point to use for the scalar. `v` must be
-  ///   a valid Unicode scalar value, in the range `0...0xD7FF` or
-  ///   `0xE000...0x10FFFF`.
-  public init(_ v: UInt32) {
+  /// In case an invalid input value, nil is returned.
+  ///
+  ///     let codepoint: UInt32 = extValue // This might be an invalid value. 
+  ///     let emoji = UnicodeScalar(codepoint)
+  ///     if emoji != nil {
+  ///       print(emoji!)
+  ///     } else {
+  ///       // Do something else
+  ///     }
```

Comment:
I was trying to show that the value can be valid or invalid, i.e. with externalValue. If we know its not valid, we do not need to check for nil.